src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/AcasService.java

```diff
@@ -0,0 +1,117 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.service;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ObjectUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.hmcts.ecm.common.model.acas.AcasCertificate;
+import uk.gov.hmcts.ecm.common.model.acas.AcasCertificateRequest;
+import uk.gov.hmcts.ecm.common.model.helper.TribunalOffice;
+import uk.gov.hmcts.et.common.model.ccd.CaseData;
+import uk.gov.hmcts.et.common.model.ccd.DocumentInfo;
+import uk.gov.hmcts.et.common.model.ccd.items.RespondentSumTypeItem;
+
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static org.apache.commons.lang3.StringUtils.defaultIfEmpty;
+
+@Service
+@Slf4j
+public class AcasService {
+
+    private static final String OCP_APIM_SUBSCRIPTION_KEY = "Ocp-Apim-Subscription-Key";
+    private static final String NOT_FOUND = "not found";
+    private final TornadoService tornadoService;
+    private final RestTemplate restTemplate;
+
+    private final String acasApiUrl;
+    private final String acasApiKey;
+
+    public AcasService(TornadoService tornadoService, RestTemplate restTemplate,
+                       @Value("${acas.api.url}") String acasApiUrl,
+                       @Value("${acas.api.key}") String acasApiKey) {
+        this.tornadoService = tornadoService;
+        this.restTemplate = restTemplate;
+        this.acasApiUrl = acasApiUrl;
+        this.acasApiKey = acasApiKey;
+    }
+
+    public List<String> getAcasCertificate(CaseData caseData, String authToken) throws JsonProcessingException {
+        if (isNullOrEmpty(caseData.getAcasCertificate())) {
+            return List.of("ACAS Certificate cannot be null or empty");
+        }
+
+        List<String> acasCertificates = List.of(caseData.getAcasCertificate());
+        Object acasCertificateObject;
+        try {
+            acasCertificateObject = fetchAcasCertificates(acasCertificates.toArray(new String[0])).getBody();
+            if (ObjectUtils.isEmpty(acasCertificateObject)) {
+                return List.of("Error reading ACAS Certificate");
+            }
+        } catch (HttpClientErrorException errorException) {
+            return List.of("Error retrieving ACAS Certificate");
```

Comment:
Can we either log or show (preferably just log) HttpClientErrorException if it happens

Reply:
Exception logs